src/net.h

```diff
@@ -568,7 +568,7 @@ class CNode
 
     // inventory based relay
     CRollingFastFilter<4 * 1024 * 1024> filterInventoryKnown;
-    std::vector<CInv> vInventoryToSend;
+    std::vector<CInv> vInventoryToSend GUARDED_BY(cs_inventory);
     CCriticalSection cs_inventory;
     int64_t nNextInvSend;
```

Comment:
Oh, didn't see this section.  I think you need to put the definition for cs_inventory above any guarded by statements...some compilers need that order

-std::vector<CInv> vInventoryToSend;
-CCriticalSection cs_inventory;
+CCriticalSection cs_inventory;
+std::vector<CInv> vInventoryToSend GUARDED_BY(cs_inventory);

Reply:
Ok will swap